adafruit_hid/keyboard.py

```diff
@@ -19,149 +19,448 @@
+            # Press ctrl-x.
+            kbd.press_keys(Keyboard.LEFT_CONTROL, 'x')
+
+            # Or, more conveniently, use the CONTROL alias for LEFT_CONTROL:
+            kbd.press_keys(Keyboard.CONTROL, 'x')
+
+            # Press a, b, c keys all at once.
+            kbd.press_keys('abc')
+        """
+        for item in keys:
+            if isinstance(item, int):
+                self._add_keycode_to_report(item)
+            elif isinstance(item, str):
+                for char in item:
+                    keycode = self._char_to_keycode(char)
+                    # If this is a shifted char, clear the SHIFT flag and press the SHIFT key.
+                    if keycode & self.SHIFT_FLAG:
+                        keycode &= ~self.SHIFT_FLAG
+                        self._add_keycode_to_report(self.LEFT_SHIFT)
+                    self._add_keycode_to_report(keycode)
+            else:
+                raise ValueError("Argument is not a keycode or a string")
+
+            self.keyboard.send_report(self.report)
+
+    def release_keys(self, *keys):
+        """Send a USB HID report indicating that the given keys have been released.
+
+        Each argument is either a keycode or a string of ASCII characters.
+        For all the characters in a string, release the corresponding keycode.
+        If a character requires a shift key, the LEFT_SHIFT key is also released.
+        If a key that was not pressed is listed, it is ignored.
+
+        Examples:
+            # release keys LEFT_CONTROL, a, LEFT_SHIFT (implied by capital 'A'), and b.
+            kbd.release_keys(keycodes.LEFT_CONTROL, 'Ab')
+
+        """
+        for item in keys:
+            if isinstance(item, int):
+                self._remove_keycode_from_report(item)
+            elif isinstance(item, str):
+                for char in item:
+                    keycode = self._char_to_keycode(char)
+                    # If this is a shifted char, clear the SHIFT flag and release the Shift key.
+                    if keycode & self.SHIFT_FLAG:
+                        keycode &= ~self.SHIFT_FLAG
+                        self._remove_keycode_from_report(self.LEFT_SHIFT)
+                    self._remove_keycode_from_report(keycode)
+            else:
+                raise ValueError("Argument is not a keycode or a string.")
+
+            self.keyboard.send_report(self.report)
+
+    def release_all(self):
+        """Release all pressed keys."""
+        for i in range(8):
+            self.report[i] = 0
+        self.keyboard.send_report(self.report)
+
+    def type(self, *items):
+        """Simulate typing. Send characters and collections of keycodes as separate keypresses.
+
+        Examples:
+            # Send "abcDEF" as a sequence of separate keypresses.
+            kbd.type("abc", "DEF")
+
+            # Press and release ctrl-B, then "h", then "i", then backspace.
+            kbd.type((Keyboard.CONTROL, 'b'), "hi", Keyboard.BACKSPACE)
```

Comment:
That's fine, and see the next reply.

Side question: In Micropython, does using a `*args`  argument do extra storage allocation to build a tuple or anything like that? I am trying to avoid generating garbage.